tests/test_history.py

```diff
@@ -557,6 +558,277 @@ def handle_data(context, data):
+        # 2006-03-21 21:00:00    180.156620
+        # 2006-03-22 21:00:00    221.344654
+
+        #       When called within BTS
+        # ---------------------------------
+        # 2006-03-17 21:00:00           NaN
+        # 2006-03-20 21:00:00    139.369469
+        # 2006-03-22 00:00:00    180.156620
+
+        # Get relevant Timestamps for the history() call within handle_data()
+        oldest_hd_dt = pd.Timestamp(
+            '2006-03-20 4:00 PM', tz='US/Eastern').tz_convert('UTC')
+        penultimate_hd_dt = pd.Timestamp(
+            '2006-03-21 4:00 PM', tz='US/Eastern').tz_convert('UTC')
+
+        # Get relevant Timestamps for the history() call within BTS
+        penultimate_bts_dt = pd.Timestamp(
+            '2006-03-20 4:00 PM', tz='US/Eastern').tz_convert('UTC')
+        newest_bts_dt = normalize_date(pd.Timestamp(
+            '2006-03-22 04:00 PM', tz='US/Eastern').tz_convert('UTC'))
+
+        self.assertEquals(prices_hd[oldest_hd_dt],
+                          prices_bts[penultimate_bts_dt])
+        self.assertEquals(prices_hd[penultimate_hd_dt],
+                          prices_bts[newest_bts_dt])
+
+    def test_history_in_bts_price_minutes(self):
+        """
+        Test calling history() in before_trading_start()
+        with minutely price bars.
+        """
+        algo_text = """
+from zipline.api import history
+
+def initialize(context):
+    context.first_loop = True
+
+def before_trading_start(context, data):
+    if not context.first_loop:
+        price_bts = history(bar_count=1, frequency='1m', field='price')
+        context.price_bts = price_bts
+
+def handle_data(context, data):
+    context.first_loop = False
+
+""".strip()
+
+        #      March 2006
+        # Su Mo Tu We Th Fr Sa
+        #          1  2  3  4
+        #  5  6  7  8  9 10 11
+        # 12 13 14 15 16 17 18
+        # 19 20 21 22 23 24 25
+        # 26 27 28 29 30 31
+        start = pd.Timestamp('2006-03-20', tz='UTC')
+        end = pd.Timestamp('2006-03-22', tz='UTC')
+
+        sim_params = factory.create_simulation_parameters(
+            start=start, end=end)
+
+        test_algo = TradingAlgorithm(
+            script=algo_text,
+            data_frequency='minute',
+            sim_params=sim_params,
+            env=TestHistoryAlgo.env,
+        )
+
+        source = RandomWalkSource(start=start, end=end)
+        output = test_algo.run(source)
+        self.assertIsNotNone(output)
+
+        # Get the prices recorded by history() within BTS
+        price_bts_0 = test_algo.price_bts[0]
+        price_bts_1 = test_algo.price_bts[1]
+
+        # The prices recorded by history() in BTS should
+        # be the closing price of the previous day, which are:
+        #
+        #          sid | close on 2006-03-21
+        #         ----------------------------
+        #           0  | 180.15661995395106
+        #           1  | 578.41665003444723
+
+        self.assertEquals(180.15661995395106, price_bts_0.ix[0])
```

Comment:
I'm not sure. I could do the same thing as `test_history_in_bts_price_days`, where I call `history()` in `handle_data()` and compare it with those generated in BTS. But I've deliberately avoided this, since it was not doing this which caused the bug in `next_market_minute` to emerge. This isn't real trade data - it's generated by a random walk (but we don't vary the seed, so we know what it will be.)